src/obp_accounting_sdk/_async/longrun.py

```diff
@@ -0,0 +1,303 @@
+        """Start accounting for the current job."""
+        if self._job_id is None:
+            errmsg = "Cannot send session before making a successful reservation"
+            raise RuntimeError(errmsg)
+        data = {
+            "type": self._service_type,
+            "subtype": self._service_subtype,
+            "job_id": str(self._job_id),
+            "proj_id": str(self._proj_id),
+            "status": LongrunStatus.STARTED,
+            "instances": str(self._instances),
+            "instance_type": "fargate",
+            "timestamp": get_current_timestamp(),
+        }
+        try:
+            response = await self._http_client.post(f"{self._base_url}/usage/longrun", json=data)
+            response.raise_for_status()
+        except httpx.RequestError as exc:
+            errmsg = f"Error in request {exc.request.method} {exc.request.url}"
+            raise AccountingUsageError(message=errmsg) from exc
+        except httpx.HTTPStatusError as exc:
+            status_code = exc.response.status_code
+            errmsg = f"Error in response to {exc.request.method} {exc.request.url}: {status_code}"
+            raise AccountingUsageError(message=errmsg, http_status_code=status_code) from exc
+
+        ctx = get_context("fork") if platform.system() != "Linux" else get_context()
+
+        self._heartbeat_sender_process = ctx.Process(
+            target=self._heartbeat_sender_loop,
+            daemon=True,
+        )
+        self._heartbeat_sender_process.start()
+        self._job_running = True
+
+    async def __aenter__(self) -> Self:
+        """Initialize when entering the context manager."""
+        await self._make_reservation()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> None:
+        """Cleanup when exiting the context manager."""
+        if self._heartbeat_sender_process:
+            self._heartbeat_sender_process.terminate()
+            self._heartbeat_sender_process.join()
+
+        if not self._job_running and exc_type:
+            L.warning(f"Unhandled application error {exc_type.__name__}, cancelling reservation")
+            try:
+                await self._cancel_reservation()
+            except AccountingCancellationError as ex:
+                L.warning("Error while cancelling the reservation: %r", ex)
+
+        elif not self._job_running and not exc_val:
+            errmsg = "Accounting session must be started before closing."
+            raise RuntimeError(errmsg)
+
+        elif self._job_running and exc_type:
+            # TODO: Consider refunding the user
+            try:
+                await self._finish()
+            except AccountingUsageError as ex:
+                L.error("Error while finishing the job: %r", ex)
+
+        else:
+            try:
+                L.error("Finishing the job")
```

Comment:
This log should be probably at info or debug level.

Reply:
Thanks, fixed.